themes/default/content/blog/zephyr-code-placement-and-stacks/index.md

```diff
@@ -0,0 +1,140 @@
+    - aaron-kao
+
+# At least one tag is required. Lowercase, hyphen-delimited is recommended.
+tags:
+    - development-environment
+    - cloud-engineering
+
+# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md.
+# for additional details, and please remove these comments before submitting for review.
+---
+
+This is the first in a series of blog posts where we'll explore how a fictional company---Zephyr Archaeotech Emporium---uses Pulumi to manage their online retail store. In this first post, we'll explore a couple common questions that users ask when working with Pulumi; specifically, where should I store my Pulumi code? And how do I support multiple environments with Pulumi? This post will provide some guidance and recommendations around these topics, using Zephyr and their online store as the use case.
+
+<!--more-->
+
+Here are links to all the blog posts in the series: <!--The names of these posts and the order in which we publish them is mutable-->
+
+**Zephyr: Code Placement and Stacks** (this post)
+
+Zephyr: Structuring Their Pulumi Projects
+
+Zephyr: Moving Infrastructure State Between Projects
+
+Zephyr: Tying Stacks Together with Stack References
+
+Zephyr: Evolving the Application
+
+Zephyr: Adding Pulumi Deployments
+
+Zephyr: Refactoring for Reuse
+
+Zephyr: Local Testing with Pulumi
+
+## What is Zephyr?
+
+Zephyr is short for Zephyr Archaeotech Emporium, the fictional company in our scenario. <!--Aaron, need details on the story/background here-->
+
+## The application: Zephyr's online store
+
+Zephyr's application is the second generation of their online store. Their first generation was a monolith that they manually deployed. It was sufficient in Zephyr's early days, but as word spread and volume increased, Zephyr found it difficult to make small, incremental changes to the store, and as a result the velocity of their development efforts slowed. In preparation for the next phase of their growth---driven by robust sales of arcane artifacts---Zephyr's team evaluated a number of different architectures that would enable them to increase development velocity and give them greater flexibility in scaling different aspects of their online store. In the end, they settled on a containerized architecture deployed to Kubernetes because some of the existing team was already familiar with these technologies.
+
+Here's a diagram of their second-generation application architecture:
+
+![Placeholder Image](meta.png)
+
+Astute readers may note that this looks pretty similar to [this application](https://github.com/aws-containers/retail-store-sample-app/); in truth, we're using a fork of the AWS Containers retail store example as the basis for our fictional scenario (many thanks to the AWS Containers DA team for their hard work here!). This is an admittedly over-engineered application, but we wanted something that would allow us to address a range of customer scenarios in this blog series.
```

Comment:
just my thoughts here - 
can we expound a bit more on what this application is and the architecture? It looks like there is a diagram to come and I saw you linked the repo. But I think it would be helpful to explain a bit more detail about the application and the architecture itself and then possibly lead into a breif explanation of the "why" behind the need for pulumi and iac. I think this would help set up a more complete background for the series of posts that are going to be delivered.

Reply:
@sean1588 Are there specific details about the application and the architecture that you feel would be particularly relevant to include?

Reply:
@scottslowe Can you break out the services and how they communicate (e.g. SNS, SQS, or synchronous HTTP API requests)?

Reply:
@scottslowe Also, what compute platforms do they run on?

Reply:
I think just a very general high-level breakdown of the architecture that summaries the complexity of it and therefore will  make the case for the need of a product like pulumi 

possibly something like this....

Zephyr is an online e-commerce store that runs in AWS. The architecture consists of a UI front-end client that communicates with a variety of backend containerized microservices, that each have a responsibility of supporting the functionalit of the ecommerce site such as..... [list the services here]. Data is persisted in DynamoDB, MySQL, Redis.... (basically just to give a very general idea of the complexity involved in the app) 

Then im thinking we can lead into something along the lines of.....

these services and their supporting infrastructure can create a complex web of dependencies between the various parts of the application and infrastructure that supports it, necessitating the need for an IaC solutiion (e.g. pulumi). This will give us the benefit of enabling the infrastructure to be defined in code, which will make it easier to manage,  reproducible state, version control support, and view changes made between deployments, state management etc. This also allows us to have consistently reproducible state across multiple environments.  Pulumi is great for this because of its stack model approach that allows us to run the same code (pulumi program) against multiple environments where the specific changes among environments are managed and defined in the stacks configuration. [then this can sort of flow into the rest of the post about code and stacks]


this is just a very rough idea of what I had in mind....

Reply:
I've added some text in 35635a95 that hopefully addresses this concern.